gcloud-java-storage/src/main/java/com/google/gcloud/storage/StorageImpl.java

```diff
@@ -123,20 +125,33 @@ public BucketInfo create(BucketInfo bucketInfo, BucketTargetOption... options) {
 
   @Override
   public BlobInfo create(BlobInfo blobInfo, BlobTargetOption... options) {
-    return create(blobInfo, new ByteArrayInputStream(EMPTY_BYTE_ARRAY), options);
+    BlobInfo updatedInfo = blobInfo.toBuilder()
+        .md5(EMPTY_BYTE_ARRAY_MD5)
+        .crc32c(EMPTY_BYTE_ARRAY_CRC32C)
+        .build();
+    return create(updatedInfo, new ByteArrayInputStream(EMPTY_BYTE_ARRAY), options);
   }
 
   @Override
-  public BlobInfo create(BlobInfo blobInfo, final byte[] content, BlobTargetOption... options) {
-    return create(blobInfo,
-        new ByteArrayInputStream(firstNonNull(content, EMPTY_BYTE_ARRAY)), options);
+  public BlobInfo create(BlobInfo blobInfo, byte[] content, BlobTargetOption... options) {
+    content = firstNonNull(content, EMPTY_BYTE_ARRAY);
+    BlobInfo updatedInfo = blobInfo.toBuilder()
+        .md5(BaseEncoding.base64().encode(Hashing.md5().hashBytes(content).asBytes()))
+        .crc32c(BaseEncoding.base64().encode(
+            Ints.toByteArray(Hashing.crc32c().hashBytes(content).asInt())))
```

Comment:
Is `Ints.toByteArray(Hashing.crc32c().hashBytes(content).asInt()))` != `Hashing.crc32c().hashBytes(content).asBytes` ?